tests/nonbonded/test_nonbonded_all_pairs.py

```diff
@@ -35,6 +35,27 @@ def test_nonbonded_all_pairs_invalid_num_params():
     assert "NonbondedAllPairs::execute_device(): expected P == N_*4, got P=6, N_*4=4" == str(e.value)
 
 
+def test_nonbonded_all_pairs_get_and_set_atom_idxs(rng):
+    """Verify that getting and setting atom idxs on the GPU implementation of NonbondedAllPairs
+    behaves as expected
+    """
+    num_atoms = 231
+    beta = 2.0
+    cutoff = 1.1
+
+    potential = NonbondedAllPairs(num_atoms, beta, cutoff)
+    gpu_pot = potential.to_gpu(np.float32).unbound_impl
+    ref_atom_idxs = gpu_pot.get_atom_idxs()
+    assert len(ref_atom_idxs) == num_atoms
+    assert num_atoms == gpu_pot.get_num_atom_idxs()
+    assert len(set(ref_atom_idxs)) == len(ref_atom_idxs)
+    identity_idxs = np.arange(0, num_atoms, dtype=np.int32)
+    updated_idxs = rng.choice(identity_idxs, size=100, replace=False)
+    gpu_pot.set_atom_idxs(updated_idxs)
+    assert len(updated_idxs) == gpu_pot.get_num_atom_idxs()
+    np.testing.assert_array_equal(gpu_pot.get_atom_idxs(), updated_idxs)
```

Comment:
Should we test edge cases here? (`atom_idxs` empty, singleton, having duplicates, etc.?)

Reply:
All of those edge cases, if I understand what you mean, are verified by the atom idxs that we have for the NonbondedAllPairs. You aren't able to configure the potential with those indices and if local MD somehow did provide bad indices, the coordinates wouldn't agree. 

If you mean to test that local MD works under all of these conditions, I think that becomes hairy because the potentials need to be setup with care. At least I convinced myself that you couldn't just randomly select particles as part of the all pairs (notably you can't have the ixn group have any ligand atoms else double counts forces with precomputed). 

This care required to setup the system is definitely a weakness of how local MD functions and I would like to determine a solution long term, but in the short term this unblocks https://github.com/proteneer/timemachine/pull/1034

Reply:
I meant specifically for the `set_atom_idxs` method, but looking closer, it doesn't seem like there is any runtime validation to test here (maybe this is deliberate, since this is intended to run in a hot loop?)

Reply:
Good catch, will expand the `set_atom_idxs` tests to ensure there is validation.